src/includes/errors.rs

```diff
@@ -1,5 +1,7 @@
-#[derive(Clone, Copy, PartialEq)]
-#[cfg_attr(feature = "debug", derive(Debug))]
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
```

Comment:
`Debug` is needed for `std::error::Error` so it is explicit here.

Reply:
By the way why `#[derive(Debug)` is behind a feature flag?

Reply:
on embedded we try to not have the formatting stuff included unless absolutely necessary